### PR TITLE
Project: Add deprecation and removal warning to Advanced Settings

### PR DESCRIPTION
**Pull request by benjaoming**
image

Ref: https://github.com/readthedocs/readthedocs.org/issues/10386 https://github.com/readthedocs/readthedocs.org/issues/10352 https://github.com/readthedocs/readthedocs.org/issues/10372

### CODE REVIEW

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-    These settings can be configured using a
-    <a href="{{ config_file_link }}">configuration file</a>.
-    That's the recommended way to set up your project.
-    <strong>Settings listed here are ignored when using a configuration file.</strong
+  {% blocktrans trimmed with config_file_link="https://docs.readthedocs.io/page/config-file/v2.html" deprecation_link="https://blog.readthedocs.com/migrate-configuration-v2/" %}
+    Usage of the below settings is <strong>deprecated</strong> and support for these fields will be removed on <strong>September 25th, 2023</strong>.
+    <br><br>
+    For more information, see our blog post:
+    <a href="{{ deprecation_link }}" target="_blank">migrating your configuration to .readthedocs.yaml</a>
+    or refer to our documentation's <a href="{{ config_file_link }}" target="_blank">configuration file reference</a>.
```
Comment (humitos) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html:8]:
```suggestion
    For more information, see our blog post:
    <a href="{{ deprecation_link }}" target="_blank">migrating your configuration to .readthedocs.yaml</a>.
```

I would point all the users to the same point. The blog post is the place we decided to be the canonical one.

**Review by humitos — APPROVED** [review]
Looks good to me. I think we need to remove the link to the config file, tho. That way we point all the users to the same place avoiding confusions.

File: readthedocs/templates/projects/project_advanced_settings_helptext.html (hunk 1 above)
Comment (benjaoming) [reply readthedocs/templates/projects/project_advanced_settings_helptext.html:8]: Sounds good :+1: That was actually my first take on this, but I couldn't guess if the outcome would them be someone asking me to put it back :innocent:

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {% load i18n %}
 <p class="empty">
-  {% blocktrans trimmed with config_file_link="https://docs.readthedocs.io/page/config-file/v2.html" %}
-    These settings can be configured using a
-    <a href="{{ config_file_link }}">configuration file</a>.
-    That's the recommended way to set up your project.
-    <strong>Settings listed here are ignored when using a configuration file.</strong
+  {% blocktrans trimmed with config_file_link="https://docs.readthedocs.io/page/config-file/v2.html" deprecation_link="https://blog.readthedocs.com/migrate-configuration-v2/" %}
```
Comment (benjaoming) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html:3]:
```suggestion
  {% blocktrans trimmed with deprecation_link="https://blog.readthedocs.com/migrate-configuration-v2/" %}
```

**Pull request merged**